controllers/azuremachine_controller_test.go

```diff
@@ -714,29 +758,45 @@ func TestConditions(t *testing.T) {
 				Spec: infrav1.AzureClusterSpec{
 					AzureClusterClassSpec: infrav1.AzureClusterClassSpec{
 						SubscriptionID: "123",
+						IdentityRef: &corev1.ObjectReference{
+							Name:      "fake-identity",
+							Namespace: "default",
+							Kind:      "AzureClusterIdentity",
+						},
 					},
 				},
 			}
+			azureClusterIdentity := getFakeAzureClusterIdentity()
+			defaultSecret := &corev1.Secret{}
+
 			initObjects := []runtime.Object{
 				cluster,
 				tc.machine,
 				azureCluster,
 				tc.azureMachine,
+				azureClusterIdentity,
+				defaultSecret,
 			}
-			client := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(initObjects...).Build()
+			fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(initObjects...).Build()
+			resultIdentity := &infrav1.AzureClusterIdentity{}
+			key := client.ObjectKey{Name: azureClusterIdentity.Name, Namespace: azureClusterIdentity.Namespace}
+			g.Expect(fakeClient.Get(context.TODO(), key, resultIdentity))
 			recorder := record.NewFakeRecorder(10)
 
-			reconciler := NewAzureMachineReconciler(client, recorder, reconciler.DefaultLoopTimeout, "")
+			reconciler := NewAzureMachineReconciler(fakeClient, recorder, reconciler.DefaultLoopTimeout, "")
 
 			clusterScope, err := scope.NewClusterScope(context.TODO(), scope.ClusterScopeParams{
-				Client:       client,
+				AzureClients: scope.AzureClients{
+					Authorizer: autorest.NullAuthorizer{},
```

Comment:
Is this line required? I ask because `azure/go-autorest` is unsupported, and in trying to fix #2974 I found the use of `NullAuthorizer` generally to be dead code that I could remove and still have the tests pass.

Reply:
Looks like removing it had no effect! Nice catch.